### PR TITLE
Add options for specifying the name of the generated directory, and for exporting the response and variables types.

### PR DESCRIPTION
**Pull request by jaredly**
## Summary:
These are required for replacing gqlgen in webapp!

```ts
    // Specify the name of the generated types directory
    generatedDirectory: string = '__generated__',

    // The default generated type contains both the types of the response
    // and the variables, combined as [operatioName]Type. Setting
    // `splitTypes = true` adds the additional exports [operationName]
    // (for the response) and [operationName]Variables (for the variables).
    splitTypes?: boolean,
```

## Test plan:
see tests!

### CODE REVIEW

File: src/__test__/generateTypeFileContents.test.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+const [_, exampleSchema] = getSchemas(__dirname + '/example-schema.graphql');
+
+describe('generateTypeFileContents', () => {
+    it('split types should export response & variables types', () => {
+        const {indexContents, files} = generateTypeFileContents(
+            'hello.js',
+            exampleSchema,
+            gql`
+                query Hello {
```
Comment (kevinbarabash) [inline src/__test__/generateTypeFileContents.test.js:16]: Most of the operation names in webapp start with a lowercase letter.  I assume that that works as expected so I'm sure it's fine to leave this test as is.

File: src/__test__/generateTypeFileContents.test.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                response: {|
+
+              /** A human character*/
+              human: ?{|
+                id: string
+              |}
+            |}
+            |};
+            export type Hello = HelloType['response'];
+            export type HelloVariables = HelloType['variables'];
```
Comment (kevinbarabash) [inline src/__test__/generateTypeFileContents.test.js:54]: Nice!  I understand why the option is called `splitTypes`.

File: src/__test__/graphql-flow.test.js [hunk 3]
```diff
@@ -54,6 +54,32 @@ describe('graphql-flow generation', () => {
...
+            |},
+                response: {|
+
+              /** A human character*/
+              human: ?{|
+                id: string
+              |}
+            |}
+            |};
+        `);
```
Comment (kevinbarabash) [inline src/__test__/graphql-flow.test.js:80]: Is this an issue with jest's snapshotting or is the indentation actually a bit wonky?  No action req'd, just curious.

File: src/generateTypeFiles.js [hunk 4]
```diff
@@ -29,23 +31,15 @@ const indexPrelude = (regenerateCommand?: string) => `// @flow
...
-
-    if (!fs.existsSync(generatedDir)) {
-        fs.mkdirSync(generatedDir, {recursive: true});
-
-        // Now write an index.js for each __generated__ dir.
-        fs.writeFileSync(indexFile, indexPrelude(options.regenerateCommand));
-    }
-
-    let indexContents = fs.readFileSync(indexFile, 'utf8');
+    generatedDir: string,
```
Comment (kevinbarabash) [inline src/generateTypeFiles.js:39]: Is this necessary since `generatedDir` is in the `options`?

File: src/generateTypeFiles.js [hunk 5]
```diff
@@ -82,12 +76,51 @@ export const generateTypeFiles = (
...
+    options: Options,
+) => {
+    const generatedDir = path.join(
+        path.dirname(fileName),
+        options.generatedDirectory ?? '__generated__',
+    );
+    const indexFile = path.join(generatedDir, 'index.js');
+
+    if (!fs.existsSync(generatedDir)) {
+        fs.mkdirSync(generatedDir, {recursive: true});
```
Comment (kevinbarabash) [inline src/generateTypeFiles.js:105]: Cool. I didn't realize the built-in `mkdir` had support for recursive directories.  Time to drop `mkdirp`.

File: src/generateTypeFiles.js [hunk 6]
```diff
@@ -82,12 +76,51 @@ export const generateTypeFiles = (
...
+    }
+
+    const {indexContents, files} = generateTypeFileContents(
+        fileName,
+        schema,
+        document,
+        options,
+        generatedDir,
+        fs.readFileSync(indexFile, 'utf8'),
+    );
```
Comment (kevinbarabash) [inline src/generateTypeFiles.js:118]: Good call separating coding that doesn't do any file operations from code that does.

**Review by kevinbarabash — APPROVED** [review]
LGTM.  Thanks for adding these options. 🎉

File: src/__test__/graphql-flow.test.js (hunk 3 above)
Comment (jaredly) [reply src/__test__/graphql-flow.test.js:80]: jest snapshotting 😭

**Pull request merged**